HW4_Zolotikov/protein_tool.py

```diff
@@ -0,0 +1,200 @@
+          'Y': 'TAY', 'H': 'CAY', 'Q': 'CAR', 'N': 'AAY', 'K': 'AAR', 'D': 'GAY', 'E': 'GAR', 'C': 'TGY', 'W': '(CGN or AGR)', 'R': 'AGY', 'G': 'GGN'}
+
+
+def define_polarity(seq: str) -> Dict[str, int]:
+    """
+    Counts polar and nonpolar aminoacids in aminoacid sequences.
+
+    Arguments:
+     str: sequence to count polar and nonpolar aminoacids.
+
+    Return:
+     Dict[str, int]: 
+          Dictionary with keys 'Polar', 'Nonpolar' and values of quantity of according groups in sequence. 
+    """
+    polarity_count = {'Polar': 0, 'Nonpolar': 0}
+    for aminoacid in seq:
+        if aminoacid in POLAR_AA:
+            polarity_count['Polar'] += 1
+        else:
+            polarity_count['Nonpolar'] += 1
+    return polarity_count
+
+
+def to_dna(seq: str) -> str:
+    """
+    Transforms aminoacid sequence to DNA sequence
+
+    Arguments
+    ---------
+     str: aminoacid sequence to transform to DNA sequence.
+
+    Return
+    ------
+     str: according DNA sequence.
+    """
+    sequence_dna = []
+    for aminoacid in seq:
+        sequence_dna.append(DNA_AA[aminoacid])
+    return ''.join(sequence_dna)
+
+
+#Margarita
+ABBREVIATION_THREE_TO_ONE = {'ALA':'A', 'CYS':'C', 'ASP':'D', 'GLU':'E', 'PHE':'F',
+                             'GLY':'G', 'HIS':'H', 'ILE':'I', 'LYS':'K', 'LEU':'L',
+                             'MET':'M', 'ASN':'N', 'PRO':'P', 'GLN':'Q', 'ARG':'R',
+                             'SER':'S', 'TRE':'T', 'VAL':'V', 'TRP':'W', 'TYR':'Y'}
+AMINO_ACIDS_ONE_LETTER = {'A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K',
+                          'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V',
+                          'W', 'Y'}
+AMINO_ACIDS_THREE_LETTER = {'ALA', 'CYS', 'ASP', 'GLU', 'PHE',
+                             'GLY', 'HIS', 'ILE', 'LYS', 'LEU',
+                             'MET', 'ASN', 'PRO', 'GLN', 'ARG',
+                             'SER', 'TRE', 'VAL', 'TRP', 'TYR'}
+
+import sys
+
+def change_abbreviation(seq: str) -> str:
+  """
+  Changes the amino acid abbreviation from three-letter to one-letter.
+
+  Parametrs
+  ----------
+  seq : str
+      Amino acid sequence in three-letter form.
+  Returns
+  -------
+  str
+      Amino acid sequence in one-letter form
+  
+  """
+  one_letter_seq = [ABBREVIATION_THREE_TO_ONE[amino_acid] for amino_acid in seq.split("-")]
+  return "".join(one_letter_seq)
+
+def is_correct_seq(seq: str) -> bool:
+  """
+  Check the sequence for extraneous characters.
+
+  Parametrs
+  ----------
+  seq : str
+      Amino acid sequence.
+  Returns
+  -------
+  bool
+      TRUE - if there is no extraneous characters, FALSE - if there is extraneous characters.
+  
+  """
+  unique_amino_acids = set(seq)
+  unique_amino_acids_three = set(seq.split("-"))
+  check = unique_amino_acids <= AMINO_ACIDS_ONE_LETTER or unique_amino_acids_three <= AMINO_ACIDS_THREE_LETTER
+  return check
```

Comment:
Проблема с отступами;
советую использовать методы множеств 
```suggestion
  unique_amino_acids = set(seq)
  unique_amino_acids_three = set(seq.split("-"))
  check = unique_amino_acids.issubset(AMINO_ACIDS_ONE_LETTER) or unique_amino_acids.issubset(unique_amino_acids_three) <= AMINO_ACIDS_THREE_LETTER
  return check
```